File: py/algorithm/quick_sort.py
# quick sort
def quick_sort(sort_list):
    if len(sort_list) == 0:
        return []
    base_item = sort_list[0]
    lesser = []
    getter = []
    for idx in range(1, len(sort_list)):
        cur_item = sort_list[idx]
        if cur_item < base_item:
            lesser.append(cur_item)
        else:
            getter.append(cur_item)
    # python中，数组之间使用加号，即可将两个数组拼接成一个
    return quick_sort(lesser) + [base_item] + quick_sort(getter)

File: js/algorithm/quick_sort.js
/**
 * 快速排序算法
 * @param {Array} list 
 * 选择一个基准值，每次都跟这个基准值做比较
 */
function quickSort(list) {
    if (list.length == 0) {
        return [];
    }
    const baseItem = list[0];
    const ltArr = [];
    const gtArr = [];
    // 比基准值小的，push到ltArr，大的push到gtArr
    for (let i = 1; i < list.length; i++) {
        const curItem = list[i];
        if (curItem < baseItem) {
            ltArr.push(curItem)
        } else {
            gtArr.push(curItem)
        }
    }
    // 然后将数组合并，重复迭代
    return [...quickSort(ltArr), baseItem, ...quickSort(gtArr)]
}

module.exports = quickSort